common/spec/dependabot/git_commit_checker_spec.rb

```diff
@@ -292,20 +294,7 @@
       end
 
       context "with source code not hosted on GitHub" do
-        before do
-          allow_any_instance_of(DummyPackageManager::MetadataFinder)
-            .to receive(:look_up_source)
-            .and_return(Dependabot::Source.from_url(source_url))
-          stub_request(:get, service_pack_url)
-            .to_return(
-              status: 200,
-              body: fixture("git", "upload_packs", upload_pack_fixture),
-              headers: {
-                "content-type" => "application/x-git-upload-pack-advertisement"
-              }
-            )
-        end
-
+        let(:second_instance) { instance_double(DummyPackageManager::MetadataFinder) }
```

Comment:
It's not clear to me why you're using the `first_instance` and `second_instance` variable names?

Reply:
in this case, the names were arbitrary and the stubbing of the method meant that a new instance had to be created for each expected return value.  If I follow the above suggestions, this should not be necessary and I can use the same instance to get the different results.